common/src/main/java/com/wynntils/features/utilities/ExperienceGainMessageFeature.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © Wynntils 2022-2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.utilities;
+
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.consumers.features.Feature;
+import com.wynntils.core.notifications.MessageContainer;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.config.Category;
+import com.wynntils.core.persisted.config.Config;
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.characterstats.event.CombatXpGainEvent;
+import com.wynntils.models.profession.event.ProfessionXpGainEvent;
+import com.wynntils.models.profession.type.ProfessionType;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import java.util.EnumMap;
+import java.util.Map;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.UTILITIES)
+public class ExperienceGainMessageFeature extends Feature {
```

Comment:
Sorry for coming in late in the review, but we (or at least I) have strived for using the term "XP" consistently, instead of using "experience". (In Java classes, this means using the camel-case form `Xp`.)

I think it would be great if we stayed that way, otherwise there will be an endless confusion if stuff is named "experience" or "xp".

This comments applies to all changed code in this PR that uses "experience" instead of "xp".